File: cafe-api/Entities/Models/Product.cs
namespace Entities.Models
{
    public class Product
    {
        public int Id { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public String Name { get; set; } = null!;
        public String? Description { get; set; }
        public decimal Price { get; set; }
        public String ImageUrl { get; set; } = null!;  
    }
}

File: cafe-web/src/types/product.ts
export default interface Product {
    id: number;
    name: string;
    description?: string;
    price: number;
    imageUrl?: string;
    categoryId: number;
    categoryName: string;
    image?: File;
}

export interface ProductStats {
    id: number,
    name: string,
    count: number,
    value?: number,
}

export interface ProductAnalysis {
    lastMonthTopSoldProduct: ProductStats;
    currentMonthTopSoldProduct: ProductStats;
    lastMonthTopEarningProduct: ProductStats;
    currentMonthTopEarningProduct: ProductStats;
    suggestions: string[];
}